### PR TITLE
Remove always true if statement

### PR DESCRIPTION
**Pull request by pedrocunial**
Since `isStateful` is a function, it will always be truthy, this PR simplifies an if-statement related to it.

### CODE REVIEW

File: packages/enzyme-adapter-react-16/src/ReactSixteenAdapter.js [hunk 1]
```diff
@@ -641,29 +641,27 @@ class ReactSixteenAdapter extends EnzymeAdapter {
             ));
           }
 
-          if (isStateful) {
```
[the remark this answers is not in the document]
Comment (pedrocunial) [reply packages/enzyme-adapter-react-16/src/ReactSixteenAdapter.js]: Truth be said, this was my first contact with this code, so I'm not a 100% on the full context of this (even though I'm opening a PR in this code block).

By looking at the context, however, I agree that this really should be `isStateful(Component)`, so I'm updating to this!

**Comment by ljharb** [comment]
The change is now great - but I think you've found a bug, and before fixing it, we should add a test to prevent it from getting broken again.

**Comment by ljharb** [comment]
@pedrocunial any interest in completing this PR with a regression test?

**Comment by lamhieu-vk** [comment]
I searched `if (isStateful) {`. I think you should check and update all.
image

**Comment by ljharb** [comment]
@lamhieu-vk it's used correctly in those other places.

**Comment by lamhieu-vk** [comment]
I checked and you were right! thanks @ljharb

**Comment by ljharb** [comment]
@pedrocunial are you no longer interested in completing this PR?

Either way, I'd prefer it stay open so it can be coopted by someone else.

**Comment by pedrocunial** [comment]
@ljharb, yeah, sorry I'm not really interested in continuing this. Sorry for the whole confusion and also for closing the PR itself.

**Comment by ljharb** [comment]
I believe this was subsumed by #2476.

**Pull request closed without merging**